src/software/ai/hl/stp/play/free_kick/free_kick_play.cpp

```diff
@@ -0,0 +1,31 @@
+#include "software/ai/hl/stp/play/free_kick/free_kick_play.h"
+
+#include "shared/constants.h"
+#include "software/util/generic_factory/generic_factory.h"
+
+FreeKickPlay::FreeKickPlay(TbotsProto::AiConfig config)
+    : Play(config, true), fsm{FreeKickPlayFSM{config}}, control_params{}
+{
+}
+
+void FreeKickPlay::getNextTactics(TacticCoroutine::push_type &yield,
+                                  const WorldPtr &world_ptr)
+{
+    // This function doesn't get called and it will be removed once coroutines are phased
+    // out
+}
+
+void FreeKickPlay::updateTactics(const PlayUpdate &play_update)
+{
+    fsm.process_event(FreeKickPlayFSM::Update(control_params, play_update));
```

Comment:
FreeKickPlayFSM shouldn't be in control of all the robots and do its own defense behavior. Delegate some of the robots to DefensePlay and remove defensive behaviors from FreeKickPlayFSM

Take a look at `offense_play.cpp` for an example

Reply:
Good call! I've updated it.